### ISSUE TITLE
Question about learning viscosity

### ISSUE THREAD

**Issue by duytrangiale** [intel-isl/DeepLagrangianFluids#26]
Hi Benjamin,

Could you please describe more details on how did you train the network to learn different viscosity? The results in the paper showed that the model can predict different behaviors for different viscosity but it is not clear how to get that results. Thanks for your help.

Cheers,
Duy

**Comment by benjaminum** [comment]
In that case the viscosity parameter becomes an input feature for each particle. It also needs specialized training data that covers a range of viscosity parameters.

**Comment by duytrangiale** [comment]
Hi Benjamin,

Thanks for your suggestion. I just wonder how the kernel can learn the viscosity feature from data. I thought that it was designed to find the spatial correlation like position or velocity but this one is viscosity. Could you please explain a little bit? So if that possible, can I add more features in addition to velocity and position and the model can automatically learn it?

Thanks

**Comment by benjaminum** [comment]
Yes, the viscosity is just another input feature similar to the velocity. The network then learns from data how the viscosity changes the particle movement.

**Comment by duytrangiale** [comment]
Thanks for your suggestion, Benjamin.

**Comment by duytrangiale** [comment]
Hi Benjamin,

Could you please describe the process that you backpropagate through the network and optimize viscosity using gradient descent? How did you set up the loss function for that case? How can you get the estimated viscosity from prediction?

Thanks

**Comment by benjaminum** [comment]
The viscosity is an input for each particle and the value doesn't change (no backpropagation to the viscosity value). The loss function is the same. The only thing that changes is the data which should contain examples of fluids simulated with different viscosity parameters.

**Comment by duytrangiale** [comment]
Hi Benjamin,
Thanks for your suggestion.
